src/prefs/WaveformSettings.h

```diff
@@ -81,4 +89,34 @@ class AUDACITY_DLL_API WaveformSettings : public PrefsListener
    // Convenience
    bool isLinear() const { return stLinear == scaleType; }
 };
+
+class AUDACITY_DLL_API WaveformSettingsCache
+   : public ClientData::Cloneable< ClientData::UniquePtr >
+{
+public:
+
+   //! Get either the global default settings, or the track's own if previously created
+   static WaveformSettingsCache &Get( WaveTrack &track );
+
+   //! @copydoc Get
+   static const WaveformSettingsCache &Get( const WaveTrack &track );
+
+   ~WaveformSettingsCache() override;
+   PointerType Clone() const override;
+
+   int ZeroLevelYCoordinate(wxRect rect) const;
+
+   void GetDisplayBounds(float &min, float &max) const
+   { min = mDisplayMin; max = mDisplayMax; }
+
+   void SetDisplayBounds(float min, float max) const
```

Comment:
Why setters are const?

Reply:
See that the corresponding deleted member functions of WaveTrack were also like that.  The members were mutable.

Mutability makes some sense -- the graphics code wants to decorate a track, that it can examine but not change, with some extra memos only for its own use.

Instead of making those members of WaveTrackCache mutable, I could do this:

```
WaveformSettings &WaveformSettings::Get( const WaveTrack &track )
 {
+   return Get(const_cast<WaveTrack&>(track));
 }
```

-- that is, overload the static function accessing the cache, so it can take a const track but give back a non-const cache.